dts/x86/ia32.dtsi

```diff
@@ -27,10 +27,10 @@
 		#interrupt-cells = <3>;
 	};
 
-	sram0: memory@100000 {
+	sram0: memory@0 {
```

Comment:
> Do we really need that zero there?

If I remove it I get:

```
qemu_x86_64.dts.pre.tmp:24.16-28.4: ERROR (unit_address_vs_reg): /memory: node has a reg or ranges property, but no unit name
```

I only know enough about DTS to be dangerous. I added @galak as a reviewer in case I am not doing this the best way.

>  I discovered recently that PCI devices have a similar wart

Maybe related: we use overloaded fields for bdf and ID. See https://github.com/zephyrproject-rtos/zephyr/issues/26109

Reply:
> what looks to all the world like a memory address (which is often zero!) sitting up there in the header and have the ACTUAL address (which is not actually zero!) be found in a "register" inside it.

The reg addr is also 0, it's just 0 padded a bunch. I can make it a single 0 if that's clearer.